onnx/backend/test/case/node/rmsnormalization.py

```diff
@@ -0,0 +1,172 @@
+# Copyright (c) ONNX Project Contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+from __future__ import annotations
+
+import numpy as np
+
+import onnx
+from onnx.backend.test.case.base import Base
+from onnx.backend.test.case.node import expect
+
+
+# RMS normalization's reference implementation
+def _rms_normalization(X, W, axis=-1, epsilon=1e-5):  # type: ignore
+    X_shape = X.shape
+    X_rank = len(X_shape)
+    if axis < 0:
+        # If axis = -1 and rank of X is 4,
+        # the axis is changed to -1 + 4 = 3,
+        # which means the last axis.
+        axis = axis + X_rank
+    unsqueezed_rank = X_rank - axis
+    reduction_shape = X_shape[0:axis] + (1,) * unsqueezed_rank
+
+    # Parameter used to convert N-D tensor layer
+    # normalization to equivalent 2-D matirx operations.
+    row_number = 1
+    col_number = 1
+    for i in range(X_rank):
+        if i < axis:
+            row_number *= X_shape[i]
+        else:
+            col_number *= X_shape[i]
+
+    # After reshaping input tensor X into a matrix,
+    # layer normalization is equivalent to conducting
+    # standardization on each column vector (s.t. each
+    # column has zero mean and unit variance).
+    x_mat = np.reshape(X, (row_number, col_number))
+    # This computes mean for every x_mat's column.
+    x_mean = np.sum(x_mat, axis=1, keepdims=True) / col_number
+    x_squared_mean = np.power(x_mean, 2)
+    # This computes RMS for every x_mat's column.
+    rms = np.sum(x_squared_mean, axis=1, keepdims=True) / col_number
+    rms_plus_epsilon = rms + epsilon
+    std_dev = np.sqrt(rms_plus_epsilon)
```

Comment:
These terms, like std_dev, borrowed from layer-norm are misleading here, since we are not computing the standard deviation. Instead, this is actually rms, albeit adjusted by epsilon.